src/cli/java/org/commcare/util/cli/ApplicationHost.java

```diff
@@ -481,8 +482,13 @@ private void syncAndReport() {
 
     public void performCasePurge(UserSandbox sandbox) {
         printStream.println("Performing Case Purge");
-        CasePurgeFilter purger = new CasePurgeFilter(sandbox.getCaseStorage(),
-                SandboxUtils.extractEntityOwners(sandbox));
+        CasePurgeFilter purger = null;
+        try {
+            purger = new CasePurgeFilter(sandbox.getCaseStorage(),
+                    SandboxUtils.extractEntityOwners(sandbox));
+        } catch (InvalidCaseGraphException e) {
+            printStream.println(e.getMessage());

```

Comment:
Small thing, but I think it would be good to bail out entirely here (just `return`), since the purger will come back from this null and might cause problems downtream.

Reply:
Thanks for pointing. Created a new [PR](https://github.com/dimagi/commcare-core/pull/1005) to `formplayer` since that seem mostly relevant to web apps.